### PR TITLE
Day 15 Solution: Dueling Generators

### PR DESCRIPTION
**Pull request by shnako**
## Summary
- Implement efficient linear congruential generator algorithm for comparing pseudo-random number sequences
- Part 1: Compare lowest 16 bits of generator outputs across 40 million iterations
- Part 2: Filter generators by divisibility criteria (4 for A, 8 for B) before comparing 5 million pairs

## Test plan
- [x] Verify generator produces expected sequence from puzzle examples
- [x] Confirm lowest 16-bit matching logic works correctly
- [x] Test Part 1 with example input (588 matches) and actual input (597 matches)
- [x] Test Part 2 with filtered generators and example input (309 matches) and actual input (303 matches)
- [x] Run comprehensive test suite covering edge cases and validation

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

- **New Features**
  - Added Day 15 (Dueling Generators) support; included in the full run.
  - Added input data to enable Day 15 execution.

- **Documentation**
  - Updated progress table to include Day 15.
  - Added Day 15 puzzle description for reference.

- **Tests**
  - Added tests validating Day 15 parsing, generator behavior, and both part solutions.

### CODE REVIEW

**Comment by shnako** [comment]
@coderabbitai review

**Comment by shnako** [comment]
@coderabbitai review

File: src/main.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@ use anyhow::Result;
 use std::env;
 use std::time::Instant;
 
-const MAX_DAY: u32 = 14;
+const MAX_DAY: u32 = 15;
```
[the remark this answers is not in the document]
Comment (shnako) [reply src/main.rs:10]: Thank you for the verification! The missing functions have been successfully implemented in commit 0800988. Both solve_part1 and solve_part2 are now properly defined with the correct signatures (pub fn solve_part1(input: &str) -> u32 and pub fn solve_part2(input: &str) -> u32) and the module is correctly wired in main.rs.

File: src/solutions/day15/mod.rs [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        assert_eq!(a_val & mask, b_val & mask);
+        assert_eq!(a_val & mask, 0b1110001101001010); // Binary value from example
+    }
+    
+    #[test]
+    fn test_part1_input() {
+        let input = "Generator A starts with 516\nGenerator B starts with 190";
+        assert_eq!(solve_part1(input), 597);
+    }
+
```
[the remark this answers is not in the document]
Comment (shnako) [reply src/solutions/day15/mod.rs]: Thank you for the performance optimization suggestion\! I've addressed this in commit 0800988 by adding the #[ignore] attribute to the test_part1_input test. The test now includes the annotation to skip it during normal CI runs while still allowing it to be run locally with 'cargo test -- --ignored'. This prevents CI timeouts from the 40 million iterations.

File: src/solutions/day15/mod.rs [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        assert_eq!(gen_a.next_with_criteria(4), 740335192);
+        assert_eq!(gen_b.next_with_criteria(8), 412269392);
+    }
+    
+    #[test]
+    fn test_part2_example() {
+        // Example from puzzle: after 5 million pairs, should find 309 matches
+        let example_input = "Generator A starts with 65\nGenerator B starts with 8921";
+        assert_eq!(solve_part2(example_input), 309);
+    }
```
[the remark this answers is not in the document]
Comment (shnako) [reply src/solutions/day15/mod.rs:206]: Thank you for catching this\! I've implemented your suggestion in commit 0800988. The test_part2_example test now has the #[ignore] attribute to skip it during CI runs while preserving it for local testing. This prevents the 5 million iteration test from slowing down CI pipelines.

File: src/solutions/day15/mod.rs [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    
+    #[test]
+    fn test_part2_input() {
+        let input = "Generator A starts with 516\nGenerator B starts with 190";
+        assert_eq!(solve_part2(input), 303);
+    }
```
[the remark this answers is not in the document]
Comment (shnako) [reply src/solutions/day15/mod.rs:212]: Thank you for the consistency check\! I've addressed this in commit 0800988 by adding the #[ignore] attribute to the test_part2_input test as well. This keeps the CI fast by skipping the 5 million iteration test during normal runs, while preserving the ability to run it locally with 'cargo test -- --ignored' for validation.
